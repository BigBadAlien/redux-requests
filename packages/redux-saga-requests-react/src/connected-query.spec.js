import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ConnectedQuery from './connected-query';

const mockStore = configureStore();

describe('ConnectedQuery', () => {
  it('maps requestSelector to query', () => {
    const component = renderer.create(
      <Provider
        store={mockStore({
          request: { data: 'data', error: null, pending: 0 },
        })}
      >
        <ConnectedQuery requestSelector={state => state.request}>
          {({ data }) => <div>{data}</div>}
        </ConnectedQuery>
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
