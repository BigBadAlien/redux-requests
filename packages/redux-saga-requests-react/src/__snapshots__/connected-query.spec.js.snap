// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedQuery maps requestSelector to query 1`] = `
<div>
  data
</div>
`;
